### PR TITLE
[Example] SEAL+NGNN for ogbl-citation2

### PR DESCRIPTION
**Pull request by Ereboas**
## Checklist
Please feel free to remove inapplicable items for your PR.
- [x] The PR title starts with [$CATEGORY] (such as [NN], [Model], [Doc], [Feature]])
- [x] Changes are complete (i.e. I finished coding on this PR)
- [x] All changes have test coverage
- [x] Code is well-documented
- [x] To the best of my knowledge, examples are either not affected by this change,
      or have been fixed to be compatible with this change

## Changes

- [x] Add support for ogbl-citation2, and give ogb results for it.

### CODE REVIEW

File: examples/pytorch/ogb/ngnn_seal/models.py [hunk 1]
```diff
@@ -59,24 +62,25 @@ def __init__(
         self.convs = ModuleList()
         initial_channels = hidden_channels + self.feature_dim
 
+        self.num_ngnn_layers = num_ngnn_layers
         if ngnn_type in ["input", "all"]:
             self.convs.append(
-                NGNN(initial_channels, hidden_channels, hidden_channels)
+                NGNN(initial_channels, hidden_channels, hidden_channels, self.num_ngnn_layers)
```
Comment (frozenbugs) [inline examples/pytorch/ogb/ngnn_seal/models.py:68]: line too long.

**Comment by mufeili** [comment]
@dgl-bot

File: examples/pytorch/ogb/ngnn_seal/main.py [hunk 2]
```diff
@@ -53,11 +48,16 @@ def __init__(
         else:
             self.node_features = None
 
+        if not self.dynamic:
+            self.g_list, tensor_dict = self.load_cached()
+            self.labels = tensor_dict["y"]
+            return
```
Comment (mufeili) [inline examples/pytorch/ogb/ngnn_seal/main.py:54]: What are the motivations for changing the location of this logic?

File: examples/pytorch/ogb/ngnn_seal/main.py [hunk 3]
```diff
@@ -505,8 +508,12 @@ def print_log(*x, sep="\n", end="\n", mode="a"):
...
-            _sampled_indices = range(1000)
-            #_sampled_indices = np.random.choice(
+            _sampled_indices = list(range(1000))
+            if args.dataset.startswith("ogbl-citation"):
+                _sampled_indices = _sampled_indices + list(
+                    range(len(train_dataset) - 1000, len(train_dataset))
+                )
+            # _sampled_indices = np.random.choice(
             # len(train_dataset), 1000, replace=False
             # )
```
Comment (mufeili) [inline examples/pytorch/ogb/ngnn_seal/main.py:518]: I guess we can remove the comments L516-518.

File: examples/pytorch/ogb/ngnn_seal/main.py [hunk 4]
```diff
@@ -505,8 +508,12 @@ def print_log(*x, sep="\n", end="\n", mode="a"):
 
     if 0 < args.sortpool_k <= 1:  # Transform percentile to number.
         if args.dynamic_train:
-            _sampled_indices = range(1000)
-            #_sampled_indices = np.random.choice(
+            _sampled_indices = list(range(1000))
+            if args.dataset.startswith("ogbl-citation"):
+                _sampled_indices = _sampled_indices + list(
+                    range(len(train_dataset) - 1000, len(train_dataset))
+                )
```
Comment (mufeili) [inline examples/pytorch/ogb/ngnn_seal/main.py:515]: Why don't you directly do something like

```
if args.dataset.startswith("ogbl-citation"):
    _sampled_indices = list(range(len(train_dataset)))
else:
    _sampled_indices = list(range(1000))
```

The code will be less convoluted.

File: examples/pytorch/ogb/ngnn_seal/main.py (hunk 2 above)
Comment (Ereboas) [reply examples/pytorch/ogb/ngnn_seal/main.py:54]: Even if we don't use dynamic dataset, we should set self.edge_weights and self.node_features for the subsequent use.

**Comment by mufeili** [comment]
@dgl-bot

**Review by mufeili — APPROVED** [review]

**Comment by mufeili** [comment]
@dgl-bot

**Comment by mufeili** [comment]
Any clue for Distributed Torch CPU Unit test? @Rhett-Ying

**Comment by mufeili** [comment]
@dgl-bot

**Comment by mufeili** [comment]
@dgl-bot

**Comment by mufeili** [comment]
@frozenbugs Sorry I did not notice that you left some new comments before I merged it. Perhaps @Ereboas can take a look at the comments and see if we need to address them with a new PR.

**Comment by Ereboas** [comment]
I address issues in a new PR #4776

**Pull request merged**